akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -262,14 +275,11 @@ private[remote] abstract class InboundCompression[T >: Null](
    * Add `n` occurance for the given key and call `heavyHittedDetected` if element has become a heavy hitter.
    * Empty keys are omitted.
    */
-  // TODO not so happy about passing around address here, but in incoming there's no other earlier place to get it?
-  def increment(remoteAddress: Address, value: T, n: Long): Unit = {
-    val count = cms.addObjectAndEstimateCount(value, n)
-
-    // TODO optimise order of these, what is more expensive?
-    // TODO (now the `previous` is, but if aprox datatype there it would be faster)... Needs pondering.
-    addAndCheckIfheavyHitterDetected(value, count)
-  }
+  def increment(remoteAddress: Address, value: T, n: Long): Unit =
+    if (value != "") {
```

Comment:
can't this check be done in some upper layer (Decoder, HeaderBuilder) because this is also used for actor refs?